include/xsf/expint.h

```diff
@@ -149,6 +149,9 @@ XSF_HOST_DEVICE inline double expi(double x) {
             }
         }
         ei = ga + std::log(x) + x * ei;
+    } else if (x == HUGE_VAL) {
+        // Special use-case needed because exp(inf) / inf is undefined/NaN
+        return HUGE_VAL;
```

Comment:
```suggestion
    } else if (std::isinf(x)) {
        // Special use-case needed because exp(inf) / inf is undefined/NaN
        return std::numeric_limits<double>::infinity();
```

For consistency with existing code.